frontend/appflowy_flutter/lib/user/presentation/router.dart

```diff
@@ -138,26 +151,43 @@ class SplashRoute {
     UserProfilePB userProfile,
     WorkspaceSettingPB workspaceSetting,
   ) {
-    Navigator.push(
-      context,
-      PageRoutes.fade(
-        () => HomeScreen(
-          userProfile,
-          workspaceSetting,
-          key: ValueKey(userProfile.id),
+    if (Platform.isAndroid || Platform.isIOS) {
+      Navigator.pushAndRemoveUntil<void>(
+        context,
+        MaterialPageRoute<void>(
+          builder: (BuildContext context) => MobileHomePage(
+            userProfile,
+            workspaceSetting,
+            key: ValueKey(userProfile.id),
+          ),
         ),
-        const RouteSettings(name: routerNameWelcome),
-        RouteDurations.slow.inMilliseconds * .001,
-      ),
-    );
+        // pop up all the pages until [SplashScreen]
+        (route) => route.isFirst,
```

Comment:
IIRC, you can retrieve the name of the router and check whether it corresponds to the splash screen